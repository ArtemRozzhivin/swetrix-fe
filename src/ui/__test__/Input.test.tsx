/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Input from '../Input'

describe('Input component', () => {
  it('renders label, input and hint', () => {
    const { getAllByText, getByText, getByPlaceholderText } = render(
      <Input label='Username' hint='Enter your username' placeholder='Enter your username' />,
    )
    expect(getAllByText(/username/i)).toHaveLength(2)
    expect(getByText(/enter your username/i)).toBeInTheDocument()
    expect(getByPlaceholderText(/enter your username/i)).toBeInTheDocument()
  })

  it('calls onChange callback when input value changes', () => {
    const handleChange = jest.fn()
    const { getByRole } = render(<Input label='Username' onChange={handleChange} />)
    const input = getByRole('textbox')
    fireEvent.change(input, { target: { value: 'John' } })
    expect(handleChange).toHaveBeenCalledTimes(1)
  })

  it('displays error message when error prop is truthy', () => {
    const { getByRole, getByText } = render(<Input label='Username' error='Username is required' />)
    const input = getByRole('textbox')
    fireEvent.blur(input)
    expect(getByText(/username is required/i)).toBeInTheDocument()
  })

  it('disables input when disabled prop is true', () => {
    const { getByRole } = render(<Input label='Username' disabled />)
    const input = getByRole('textbox') as HTMLInputElement
    expect(input).toBeDisabled()
  })

  it('snapshot', () => {
    const { container } = render(<Input label='Username' />)
    expect(container).toMatchSnapshot()
  })
})
