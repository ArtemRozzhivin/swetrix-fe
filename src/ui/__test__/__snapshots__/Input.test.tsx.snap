// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component snapshot 1`] = `
<div>
  <div
    class=""
  >
    <div
      class=""
    >
      <label
        class="flex text-sm font-medium text-gray-700 dark:text-gray-200"
        for=""
      >
        Username
      </label>
    </div>
    <div
      class="mt-1 relative"
    >
      <input
        aria-describedby="-optional"
        class="shadow-sm focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 dark:text-gray-50 dark:placeholder-gray-400 dark:border-gray-800 dark:bg-gray-700 rounded-md"
        id=""
        name=""
        placeholder=""
        type=""
        value=""
      />
    </div>
    <p
      class="mt-2 text-sm text-gray-500 dark:text-gray-300 whitespace-pre-line"
      id="-optional"
    />
  </div>
</div>
`;
